### PR TITLE
Added java optimized fibo

### PR DESCRIPTION
**Pull request by ikhoury**

### CODE REVIEW

**Comment by ikhoury** [comment]
Added optimized java fib class

File: FibOptimized.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        }
+    }
+    public static void main(String[] args){
+        if(args.length != 1){
+            System.err.println("Error! expected one numebr argument");
+            System.exit(1);
+        }
+        Instant start = Instant.now();
+        System.out.println("Processing...");
+        fibonnaci(Long.parseLong(args[0]));
```
Comment (drujensen) [inline FibOptimized.java]: right now, we are not using params. This is an enhancement request to do that but the other ones are not doing  this.

File: FibOptimized.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        long duration = Duration.between(start, Instant.now()).toMillis();
```
Comment (drujensen) [inline FibOptimized.java]: Please remove this.  We are using `time` from the command line. this includes load time.  I know this is kinda unfair to vm's but not sure a way around it.  I have noted loading a VM as something to consider when viewing the times.

File: FibOptimized.java (hunk 1 above)
Comment (ikhoury) [reply FibOptimized.java]: Noted, I will review other classes and follow the same implementation.

**Comment by ikhoury** [comment]
I refactored the class as discussed

**Pull request merged**